cli/flags.go

```diff
@@ -865,14 +865,39 @@ type StringSliceVar struct {
 
 func (f *FlagSection) StringSliceVar(i *StringSliceVar) {
 	parser := func(s string) ([]string, error) {
-		parts := strings.Split(s, ",")
-		for i := len(parts) - 2; i >= 0; i-- {
-			if strings.HasSuffix(parts[i], "\\") {
-				parts[i] = parts[i][:len(parts[i])-1] + "," + parts[i+1]
-				parts = append(parts[:i+1], parts[i+2:]...)
+		var parts []string
+		var b strings.Builder
+		var escaped bool
+
+		for _, r := range s {
+			if r == '\\' {
+				if escaped {
+					b.WriteRune(r)
+					escaped = false
+				} else {
+					escaped = true
+				}
+				continue
 			}
+
+			if !escaped && r == ',' {
+				escaped = false
+				if v := strings.TrimSpace(b.String()); v != "" {
+					parts = append(parts, v)
+				}
+				b.Reset()
+				continue
+			}
+
+			escaped = false
+			b.WriteRune(r)
```

Comment:
I think this might be a bit cleaner
```suggestion
			if escaped {
				escaped = false
				b.WriteRune(r)
				continue
			}
			switch r {
			case '\\':
				escaped = true
				break
			}
			case ',' {
				if v := strings.TrimSpace(b.String()); v != "" {
					parts = append(parts, v)
				}
				b.Reset()
				break
			default:
				b.WriteRune(r)
				break
			}
```